### PR TITLE
DBZ-773 Unifying some code from the relational DB schemas

### PR DESCRIPTION
**Pull request by gunnarmorling**
https://issues.jboss.org/browse/DBZ-773

@jpechane Can you check out this one? I'm gonna base the work for DBZ-644 (source type as schema parameters) on this, so it's done once and for all for all the relational connectors.

### CODE REVIEW

File: debezium-core/src/main/java/io/debezium/pipeline/EventDispatcher.java [hunk 1]
```diff
@@ -40,16 +42,20 @@
...
     private final ChangeEventQueue<Object> queue;
     private final DataCollectionFilter<T> filter;
 
-    public EventDispatcher(TopicSelector topicSelector, DatabaseSchema schema,
+    public EventDispatcher(TopicSelector<T> topicSelector, DatabaseSchema<T> schema,
             ChangeEventQueue<Object> queue,
             DataCollectionFilter<T> filter) {
         this.topicSelector = topicSelector;
         this.schema = schema;
+        this.historizedSchema = schema instanceof HistorizedRelationalDatabaseSchema
```
Comment (jpechane) [inline debezium-core/src/main/java/io/debezium/pipeline/EventDispatcher.java]: Is `HistorizedRelationalDatabaseSchema` really subtype of `HistorizedDatabaseSchema`?

**Comment by jpechane** [comment]
@gunnarmorling I think it is good, I just do not understand if there is any relation between `HistorizedDatabaseSchema` and `HistorizedRelationalDatabaseSchema`

File: debezium-core/src/main/java/io/debezium/pipeline/EventDispatcher.java (hunk 1 above)
Comment (gunnarmorling) [reply debezium-core/src/main/java/io/debezium/pipeline/EventDispatcher.java]: Could catch, I should check for HistorizedDatabaseSchema here, as this class shouldn't care wether relational or not. That said, indeed HistorizedRelationalDatabaseSchema should extend HistorizedDatabaseSchema. I'll change the two things and merge.

**Comment by gunnarmorling** [comment]
Addressed your remark, rebased and applied. Thanks for reviewing.

**Pull request closed without merging**